packages/core/src/code-gen/generateRPCProxy.ts

```diff
@@ -98,270 +126,566 @@ export default function generateRPCProxy(
-                factory.createVariableDeclarationList(
-                  [
-                    factory.createVariableDeclaration(
-                      factory.createIdentifier("output"),
-                      undefined,
-                      undefined,
-                      factory.createAwaitExpression(
-                        factory.createCallExpression(
-                          factory.createIdentifier(
-                            `${serviceName}.${funcDef.name}.func`,
-                          ),
-                          undefined,
-                          funcDef.parameters.map((param) =>
-                            factory.createPropertyAccessExpression(
-                              factory.createPropertyAccessExpression(
-                                factory.createIdentifier("inputParseResult"),
-                                factory.createIdentifier("result"),
-                              ),
-                              factory.createIdentifier(
-                                getParameterName(param.name),
-                              ),
-                            ),
-                          ),
-                        ),
-                      ),
+                    1,
+                  ),
+                }),
+                tsx.const({
+                  name: "parseMiddlewareNextOut",
+                  init: tsx.expression.elementAccess(
+                    tsx.expression.elementAccess(
+                      `service_middlewares_${service.name}`,
+                      "i",
                     ),
-                  ],
-                  ts.NodeFlags.Const |
-                    ts.NodeFlags.AwaitContext |
-                    ts.NodeFlags.ContextFlags |
-                    ts.NodeFlags.TypeExcludesFlags,
-                ),
-              ),
-              factory.createVariableStatement(
-                undefined,
-                factory.createVariableDeclarationList(
-                  [
-                    factory.createVariableDeclaration(
-                      factory.createIdentifier("outputParseResult"),
-                      undefined,
-                      undefined,
-                      factory.createCallExpression(
-                        factory.createIdentifier("outputParser"),
-                        undefined,
-                        [factory.createIdentifier("output")],
+                    2,
+                  ),
+                }),
+                tsx.const({
+                  name: "currResolverIndex",
+                  init: tsx.expression.identifier("i"),
+                }),
+                tsx.const({
+                  name: "nextResolverIndex",
+                  init: tsx.expression.binary(
+                    tsx.expression.identifier("i"),
+                    "+",
+                    tsx.literal.number(1),
+                  ),
+                }),
+
+                // const ctx = await resolvers[currResolverIndex].inputContext.promise
```

Comment:
This can be removed, right? And the other commented out things in this file as well?

Reply:
Ah right, will fix.

Reply:
ah This is actually what the statement beneath will generate. So this helps a bit reading the code :)